### DISCUSSION TITLE
high cpu load

### DISCUSSION THREAD

**Discussion by chengang-hub** [SoftEtherVPN/SoftEtherVPN#1809] [General]
Hi, I have a SoftEther vpn cluster， one master  and 2 node
Observing the monitoring data, it was found that the CPU load suddenly became high from 2022-04-24

![image](https://user-images.githubusercontent.com/108459883/227769257-e70a6416-8148-4048-9256-565b55ba4dd9.png)

On the same day, the VPN service was restarted and the public server IP of the node node was set,No other operations have been done.
![image](https://user-images.githubusercontent.com/108459883/227769406-14d4d544-a007-42bf-b764-ea882e03f34a.png)

**SoftEther version:**[SoftEther VPN Server (Ver 4.38, Build 9760, rtm)] 

**Component:** [Server]
**Operating system:** [Linux] 
**Architecture:** [64 bit]

master [HUB1]： 8 CPU ， 16G ，  users  session  1100， TCP :1500, UDP: 1100 ,lisening port  40
node  [HUB2]  ：4 CPU  8G   users  session  470 ,    TCP :650 , UDP: 480 ,lisening port  40
 
What causes the CPU load to suddenly become high? 
The recommended number of users per slave node server should not exceed the number？
 by the way   I am unable to register successfully in https://www.vpnusers.com/
thanks。

**Comment by shipitsin-ilia** [comment]
yep, it's ok to ask such questions here.

"load average"  means high number of processes involved into IO operations.

can you compare log writing activity (compared to previous day?), i.e. number of log lines per day/hour/minute ?
is there  something interesting in logs ?

**Reply by chengang-hub** [reply]
I can't find the VPN logs for 2022-02-24, because the number of users has increased a lot, the number of logs has increased a lot now, let's analyze the current logs first, It stands to reason that restarting the VPN server without a change in the number of users will not cause an increase in CPU load the next day。
2、does an increase in slave node lead to an increase in master node load? We may add more slave nodes later

thank you!

**Comment by chengang-hub** [comment]
Looking at the logs, I found that there are some possible anomalies: 
1. Several expired users are constantly trying to connect. 
![image](https://user-images.githubusercontent.com/108459883/229010931-2460786c-ae8b-4ff4-b18c-d159671833a6.png)
2、
![image](https://user-images.githubusercontent.com/108459883/229012819-1d2b1f3f-8838-449a-a9b1-383775ffc46a.png)
@shipitsin-ilia shipitsin-ilia
. Can you help me see if it may be caused by this? thank you！

**Comment by chengang-hub** [comment]
there are too many sleeping threads.Why do these threads have a CPU time of up to 8535 minutes, where is the problem
Can you help analyze the reason? Is it caused by restarting the VPN service? I haven't restarted my VPN service in 1 year, can I try doing it again? Thank you so much @chipitsine @shipitsin-ilia @SoftEtherVPN @dnobori @davidebeatrici
![image](https://user-images.githubusercontent.com/108459883/229417587-f3972ef0-8a97-432c-a95f-da656a5beb59.png)
![image](https://user-images.githubusercontent.com/108459883/229417746-ed3eadf8-6f97-4e01-94c3-c10858b88499.png)

**Reply by chengang-hub** [reply]
I found that as long as the number of threads increases by 1 after each client user connects, the number of threads will not be released even if the client user disconnects later

**Comment by chengang-hub** [comment]
I found a similar discussion and I found that this pull request was not merged into the stable version, could it be the same issue for us? Will the stable version fix this?

https://github.com/SoftEtherVPN/SoftEtherVPN/pull/258

**Comment by OmlineEditor** [comment]
I had a similar situation with a high CPU load. it lasted for about 20-30 minutes, then it stopped for no apparent reason. but I only have 3 users on the server. my SoftEtherVPN version 5.02.5187. a high load of up to 100% of the CPU appears spontaneously and it is unclear why. everything is as usual in the logs and there is no obvious problem.
